src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -705,12 +705,12 @@ private async Task Orphan(IHost instance, CancellationToken cancellationToken =
                             }
                         }, cancellationToken);
 
-                        GetHostLogger(instance).LogDebug("Standby ScriptHost marked for disposal");
+                        _logger.LogDebug("Standby ScriptHost marked for disposal");
```

Comment:
Ah, good catch. `GetHostLogger()` was meant to account for disposals -- but now I see there is a race there. Disposal could happen after we check and before we call `LogDebug()`.